lib/Controller/UsersController.php

```diff
@@ -469,6 +500,213 @@ public function create($username, $password, array $groups= [], $email='') {
+
+		$mail = new TemplateResponse('user_management', 'new_user/email-plain_text', $mailData, 'blank');
+		$plainTextMailContent = $mail->render();
+
+		$subject = $this->l10n->t('Your %s account was created', [$this->defaults->getName()]);
+
+		try {
+			$message = $this->mailer->createMessage();
+			$message->setTo([$user->getEMailAddress() => $userId]);
+			$message->setSubject($subject);
+			$message->setHtmlBody($mailContent);
+			$message->setPlainBody($plainTextMailContent);
+			$message->setFrom([$this->fromMailAddress => $this->defaults->getName()]);
+			$this->mailer->send($message);
+		} catch (\Exception $e) {
+			$this->log->error("Can't send new user mail to " . $user->getEMailAddress() . ": " . $e->getMessage(), ['app' => 'settings']);
+		}
+
+		return new TemplateResponse(
+			'user_management', 'new_user/tokensendnotify', [], 'guest'
+		);
+	}
+
+	/**
+	 * @PublicPage
+	 * @NoAdminRequired
+	 * @NoSubadminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @param $token
+	 * @param $userId
+	 * @param $password
+	 * @return array|RedirectResponse|TemplateResponse
+	 */
+	public function setPassword($token, $userId, $password) {
+		try {
+			$this->checkPasswordSetToken($token, $userId);
+			$user = $this->userManager->get($userId);
+
+			if (!$user->setPassword($password)) {
+				return new TemplateResponse(
+					'core', 'error', [
+						"errors" => [[
+							"error" =>
+								$this->l10n->t('The password can not be set for user %s. ' .
+									'Contact your admin.', [$userId])]]
+					], 'guest'
+				);
+			}
+
+			\OC_Hook::emit('\OC\Core\LostPassword\Controller\LostController', 'post_passwordReset', ['uid' => $userId, 'password' => $password]);
+			\OC_User::unsetMagicInCookie();
+		} catch (UserTokenException $e) {
+			return new TemplateResponse(
+				'core', 'error', [
+					"errors" => [["error" => $e->getMessage()]]
+				], 'guest'
+			);
+		}
+
+		try {
+			$this->sendNotificationMail($userId);
+		} catch (\Exception $e) {
+			$this->log->warning($e->getMessage());
+			return ['status' => 'error', 'msg' => $this->l10n->t('Not able to send email. Kindly check the logs')];
+		}
+
+		$this->logout();
+		return new RedirectResponse($this->urlGenerator->linkToRouteAbsolute('core.login.showLoginForm'));
+	}
+
+	/**
+	 * @param string $userId
+	 * @throws \Exception
+	 */
+	protected function sendNotificationMail($userId) {
+		$user = $this->userManager->get($userId);
+		$email = $user->getEMailAddress();
+
+		if ($email !== '') {
+			$tmpl = new \OC_Template('core', 'lostpassword/notify');
+			$msg = $tmpl->fetchPage();
+
+			try {
+				$message = $this->mailer->createMessage();
+				$message->setTo([$email => $userId]);
+				$message->setSubject($this->l10n->t('%s password changed successfully', [$this->defaults->getName()]));
+				$message->setPlainBody($msg);
+				$message->setFrom([$email => $this->defaults->getName()]);
+				$this->mailer->send($message);
+			} catch (\Exception $e) {
+				throw new \Exception($this->l10n->t(
```

Comment:
This shouldn't need to be translated

Reply:
Removed the translation of this message.